packages/vaadin-time-picker/src/vaadin-time-picker.js

```diff
@@ -59,89 +71,82 @@ import './vaadin-time-picker-text-field.js';
+) {
+  static get is() {
+    return 'vaadin-time-picker';
+  }
+
   static get template() {
     return html`
       <style>
-        :host {
-          display: inline-block;
+        /* See https://github.com/vaadin/vaadin-time-picker/issues/145 */
+        :host([dir='rtl']) [part='input-field'] {
+          direction: ltr;
         }
 
-        :host([hidden]) {
-          display: none !important;
+        :host([dir='rtl']) [part='input-field'] ::slotted(input)::placeholder {
+          direction: rtl;
+          text-align: left;
         }
 
         [part~='toggle-button'] {
           cursor: pointer;
         }
-
-        .input {
-          width: 100%;
-          min-width: 0;
-        }
       </style>
-      <vaadin-combo-box-light
-        allow-custom-value=""
-        filtered-items="[[__dropdownItems]]"
-        disabled="[[disabled]]"
-        readonly="[[readonly]]"
-        auto-open-disabled="[[autoOpenDisabled]]"
-        dir="ltr"
-        theme$="[[theme]]"
-      >
-        <vaadin-time-picker-text-field
-          class="input"
-          name="[[name]]"
-          invalid="[[invalid]]"
-          autocomplete="off"
-          label="[[label]]"
-          required="[[required]]"
+
+      <div part="container">
+        <div part="label" on-click="focus">
+          <slot name="label"></slot>
+          <span part="indicator" aria-hidden="true"></span>
+        </div>
+
+        <vaadin-time-picker-combo-box
+          id="comboBox"
+          filtered-items="[[__dropdownItems]]"
+          value="{{_comboBoxValue}}"
           disabled="[[disabled]]"
-          prevent-invalid-input="[[preventInvalidInput]]"
-          pattern="[[pattern]]"
-          error-message="[[errorMessage]]"
-          autofocus="[[autofocus]]"
-          placeholder="[[placeholder]]"
           readonly="[[readonly]]"
-          role="application"
-          aria-live="assertive"
-          min$="[[min]]"
-          max$="[[max]]"
-          aria-label$="[[label]]"
-          clear-button-visible="[[clearButtonVisible]]"
-          i18n="[[i18n]]"
-          helper-text="[[helperText]]"
+          auto-open-disabled="[[autoOpenDisabled]]"
+          position-target="[[_inputContainer]]"
           theme$="[[theme]]"
+          on-change="__onChange"
         >
-          <slot name="helper" slot="helper">[[helperText]]</slot>
-          <span
-            slot="suffix"
-            part="toggle-button"
-            class="toggle-button"
-            role="button"
-            aria-label$="[[i18n.selector]]"
-          ></span>
-        </vaadin-time-picker-text-field>
-      </vaadin-combo-box-light>
+          <vaadin-input-container
+            part="input-field"
+            readonly="[[readonly]]"
+            disabled="[[disabled]]"
+            invalid="[[invalid]]"
+            theme$="[[theme]]"
+          >
+            <slot name="prefix" slot="prefix"></slot>
+            <slot name="input"></slot>
```

Comment:
There's a UX regression: if you try to modify the input's value, a value gets automatically focused, the input value is overridden and you lose the caret position:

Before:
![tp-ux-1](https://user-images.githubusercontent.com/1222264/134125746-05e7edbb-1041-49b3-9d8d-0459c66454bf.gif)

After
![tp-ux-2](https://user-images.githubusercontent.com/1222264/134125799-60c03ed0-bbb0-4eaa-8091-16fb449ed39c.gif)
:

Reply:
Done.